examples/substrait-spark/README.md

```diff
@@ -0,0 +1,586 @@
+            +- Join:: INNER <ScalarFn>equal:any_any
+                : arg0 = FieldRef#/Str/StructField{offset=0}
+                : arg1 = FieldRef#/Str/StructField{offset=2}
+               +- Project:: [Str, Str, Str, Str, Str, Str, Str, Str, Str]
+                  +- Filter:: <ScalarFn>is_not_null:any
+                      : arg0 = FieldRef#/Str/StructField{offset=0}
+                     +- LocalFiles::
+                         :  file:///opt/spark-data/vehicles_subset_2023.csv len=1547 partition=0 start=0
+               +- Project:: [Str, Str, Str, Str, Str, Str, Str, Str, Str]
+                  +- Filter:: <ScalarFn>and:bool
+                      : arg0 = <ScalarFn>and:bool
+                         : arg0 = <ScalarFn>is_not_null:any
+                            : arg0 = FieldRef#/Str/StructField{offset=5}
+                         : arg1 = <ScalarFn>equal:any_any
+                            : arg0 = FieldRef#/Str/StructField{offset=5}
+                            : arg1 = <StrLiteral P>
+                      : arg1 = <ScalarFn>is_not_null:any
+                         : arg0 = FieldRef#/Str/StructField{offset=1}
+                     +- LocalFiles::
+                         :  file:///opt/spark-data/tests_subset_2023.csv len=1491 partition=0 start=0
+```
+
+There is a more detail in this version that the Spark versions; details of the functions called for example are included. However, the structure of the overall plan is identical with 1 exception. There is an additional `project` relation included between the `sort` and `aggregate` - this is necessary to get the correct types of the output data.
+
+We can also see in this case as the plan came from Spark directly it's also included the location of the datafiles. Below when we reload this into Spark, the locations of the files don't need to be explicitly included.
+
+
+As `Substrait Spark` library also allows plans to be loaded and executed, so the next step is to consume these Substrait plans.
+
+## Consuming a Substrait Plan
+
+The [`SparkConsumeSubstrait`](./app/src/main/java/io/substrait/examples/SparkConsumeSubstrait.java) code shows how to load this file, and most importantly how to convert it to a Spark engine plan to execute
+
+Loading the binary protobuf file is the reverse of the writing process (in the code the file name comes from a command line argument, here we're showing the hardcode file name )
+
+```java
+      byte[] buffer = Files.readAllBytes(Paths.get("spark_sql_substrait.plan"));
+      io.substrait.proto.Plan proto = io.substrait.proto.Plan.parseFrom(buffer);
+
+      ProtoPlanConverter protoToPlan = new ProtoPlanConverter();
+      Plan plan = protoToPlan.from(proto);
+
+```
+The loaded byte array is first converted into the protobuf Plan, and then into the Substrait Plan object. Note it can be useful to name the variables, and/or use the pull class names to keep track of it's the ProtoBuf Plan or the high-level POJO Plan.
```

Comment:
> and/or use the pull class names to keep track of it's the ProtoBuf Plan or the high-level POJO Plan

I'm not sure what is meant here.

Reply:
Should the `full class name`.  Added a line 

```
For example `io.substrait.proto.Plan` or `io.substrait.Plan`
```